projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/components/linked-cases-table/linked-cases-to-table.component.ts

```diff
@@ -136,6 +137,30 @@ export class LinkedCasesToTableComponent implements OnInit, AfterViewInit {
     }
   }
 
+  public sortReasonCodes(searchCasesResponse): LinkedCasesResponse[] {
+    searchCasesResponse.forEach((item: any) => {
+      if(item && item.reasons && item.reasons.length){
+        item.reasons.forEach((reason) => {
+          reason.sortOrder = this.getReasonSortOrder(reason.value.Reason)
+        });
+        item.reasons = item.reasons.sort((a, b) => a.sortOrder - b.sortOrder);
+        item.sortOrder = item.reasons[0].sortOrder;
+      }
+    });
+    searchCasesResponse = searchCasesResponse && searchCasesResponse.sort((a, b) => a.sortOrder - b.sortOrder)
+    return searchCasesResponse;
+  }
+
+  public getReasonSortOrder(reasonCode: string): number {
+    if (reasonCode === LinkedCasesToTableComponent.CASE_PROGRESS_REASON_CODE) {
```

Comment:
@Kartheek-S - This can be written as below:
```
if (reasonCode === LinkedCasesToTableComponent.CASE_PROGRESS_REASON_CODE) {
      return 1;
}
if (reasonCode === LinkedCasesToTableComponent.CASE_CONSOLIDATED_REASON_CODE) {
      return 2;
}
return 3;
```